src/Lachain.Core/Blockchain/VM/ExternalHandler.cs

```diff
@@ -378,6 +378,53 @@ public static void Handler_Env_GetMsgValue(int dataOffset)
             if (!ret)
                 throw new InvalidContractException("Bad call to (get_msg_value)");
         }
+        
+        public static void Handler_Env_GetExternalBalance(int addressOffset, int resultOffset)
+        {
+            Logger.LogInformation($"Handler_Env_GetExternalBalance({addressOffset}, {resultOffset})");
+            var frame = VirtualMachine.ExecutionFrames.Peek() as WasmExecutionFrame
+                        ?? throw new InvalidOperationException("Cannot call GetExternalBalance outside wasm frame");
+            
+            // Get the address from the given memory offset
+            var snapshot = frame.InvocationContext.Snapshot;
+            var addressBuffer = SafeCopyFromMemory(frame.Memory, addressOffset, 20);
+            if (addressBuffer is null)
+                throw new InvalidContractException("Bad call to (get_block_timestamp)");
```

Comment:
get_external_balance,  not  get_block_timestamp here.)
```suggestion
                throw new InvalidContractException("Bad call to (get_external_balance)");
```
You can accept this changes from the GitHub page directly,  by pressing the button "Commit suggestion" )